pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -248,6 +249,12 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, errors.Wrapf(err, "unable to get secret %s", request.NamespacedName)
 	}
 
+	// Initialize prometheus configuration
+	pc, err := metrics.NewPrometheusNodeConfig(r.k8sclientset)
```

Comment:
Is it possible to move this closer to the scope of its usage?

Reply:
Since the configuration is being called at different places in the controller, I am tending to keep this here

Reply:
Instead of doing this here, what about making this part of the reconciler object and doing it when creating the reconciler? The object isnt changing on each reconcile is it?